# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieChangeEvent.cpp
# Commit: 0868fb4018a7
# Full Hash: 0868fb4018a7ae2002b8f2fcfd88643c2e060f07
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-11 09:21:34
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 2 - CookieStore API - IPDL, r=edgul,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215145
# ==============================================================================

diff -r 9c8b6e35bf3f -r 0868fb4018a7 dom/cookiestore/CookieChangeEvent.cpp
--- a/dom/cookiestore/CookieChangeEvent.cpp	Tue Sep 10 16:31:38 2024 +0000
+++ b/dom/cookiestore/CookieChangeEvent.cpp	Tue Sep 10 16:31:38 2024 +0000
@@ -50,4 +50,30 @@
   return event.forget();
 }
 
+/* static */ already_AddRefed<CookieChangeEvent>
+CookieChangeEvent::CreateForChangedCookie(EventTarget* aEventTarget,
+                                          const CookieListItem& aItem) {
+  RefPtr<CookieChangeEvent> event =
+      new CookieChangeEvent(aEventTarget, nullptr, nullptr);
+
+  event->InitEvent(u"change"_ns, false, false);
+  event->SetTrusted(true);
+
+  event->mChanged.AppendElement(aItem);
+  return event.forget();
+}
+
+/* static */ already_AddRefed<CookieChangeEvent>
+CookieChangeEvent::CreateForDeletedCookie(EventTarget* aEventTarget,
+                                          const CookieListItem& aItem) {
+  RefPtr<CookieChangeEvent> event =
+      new CookieChangeEvent(aEventTarget, nullptr, nullptr);
+
+  event->InitEvent(u"change"_ns, false, false);
+  event->SetTrusted(true);
+
+  event->mDeleted.AppendElement(aItem);
+  return event.forget();
+}
+
 }  // namespace mozilla::dom